.github/workflows/bench.yml

```diff
@@ -0,0 +1,132 @@
+# Adapted from https://github.com/paradigmxyz/reth/blob/main/.github/workflows/bench.yml
+name: Bench
+
+on:
+  pull_request:
+  merge_group:
+
+env:
+  CARGO_TERM_COLOR: always
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.ref || github.run_id }}
+  cancel-in-progress: true
+
+jobs:
+  benchmark:
+    name: Bench selected tests
+    runs-on: ubuntu-latest
+    # Only run benchmarks in merge groups
+    if: github.event_name != 'pull_request'
+    steps:
+      - name: Checkout Noir repo
+        uses: actions/checkout@v3
+
+      - name: Collect locked barretenberg rev
+        run: |
+          echo "BB_REV=$(jq -r .nodes.barretenberg.locked.rev ./flake.lock)" >> $GITHUB_ENV
+          echo "BB_REV is ${{ env.BB_REV }}"
+
+      - name: Checkout barretenberg
+        uses: actions/checkout@v3
+        with:
+          repository: AztecProtocol/barretenberg
+          path: barretenberg
+          ref: ${{ env.BB_REV }}
+
+      - name: Setup Linux environment
+        run: |
+          sudo apt update
+          sudo apt install libomp-dev cmake ninja-build
+          echo "PRESET=default" >> $GITHUB_ENV
+
+      - name: Build and install barretenberg
+        working-directory: barretenberg/cpp
+        run: |
+          cmake --preset ${{ env.PRESET }} -DTESTING=OFF -DBENCHMARKS=OFF -DCMAKE_BUILD_TYPE=RelWithAssert -DDISABLE_ASM=ON -DDISABLE_ADX=ON
+          cmake --build --preset ${{ env.PRESET }}
+          sudo cmake --install build
+
+      - name: Checkout master sources
+        uses: actions/checkout@v3
+        with:
+          ref: master
+          path: master
+
+      - name: Checkout PR sources
+        uses: actions/checkout@v3
+        with:
+          clean: false
+          path: pr
+
+      - name: Setup toolchain
+        uses: dtolnay/rust-toolchain@master
+        with:
+          toolchain: 1.66.0
+
+      - name: Install Valgrind
+        run: |
+          sudo apt install valgrind
+
+      - name: Set master baseline
+        run: cargo bench --bench iai --manifest-path master/Cargo.toml
+
+      - name: Compare PR benchmark
+        shell: 'script -q -e -c "bash {0}"' # required to workaround /dev/tty not being available
+        run: |
+          ./pr/.github/scripts/compare_iai.sh
+
+  # Checks that benchmarks not run in CI compile
+  bench-check:
+    name: check
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Noir repo
+        uses: actions/checkout@v3
+
+      - name: Collect locked barretenberg rev
```

Comment:
If we don't benchmark prove and only execute, could we get rid of this setup for barretenberg?

Reply:
Can you re-use the same machinery we use in tests to install barretenberg, ie using nix? If not, its not a big deal since we are not using `nargo prove`